tests/homeWork18.spec.js

```diff
@@ -0,0 +1,162 @@
+import {expect, test} from '@playwright/test'
+test.describe('check SignUP form', ()=>{  
+    
+    test.beforeEach(async ({page})=>{
+        await page.goto("")
+        await page.locator('button', {hasText: 'Sign up'}).click()
+    })
+
+    test.describe('positive test registration', ()=>{
+        test.afterEach('Delete user',async ({page}) => {
+            
+            await page.locator('button', {hasText: 'My profile'}).click()
+            await page.locator('a[routerlink="settings"]').click()
+            await page.locator('button', {hasText: 'Remove my account'}).click()
+            await page.locator('button[class="btn btn-danger"]', {hasText: 'Remove'}).click()
+
+            await expect(page).toHaveURL("")
+        });
+
+        test("user registration", async({page})=>{
+            const popup = page.locator('app-signup-modal')
+            const fildName = popup.locator('#signupName')
+            const lastName = popup.locator('#signupLastName')
+            const email = popup.locator('#signupEmail')
+            const password = popup.locator('#signupPassword')
+            const repeatPassword = popup.locator('#signupRepeatPassword')
+            const btnRegister = page.locator('button', {hasText: 'Register'} )
+
+            await fildName.pressSequentially('TestName', {delay: 100})
+            await lastName.pressSequentially('TestLastName', {delay: 100})
+            await email.pressSequentially('aqa-test@email.com', {delay: 100})
+            await password.pressSequentially('Password1', {delay: 100})
+            await repeatPassword.pressSequentially('Password1', {delay: 100})
+            await btnRegister.click()
+            await expect(page).toHaveURL("/panel/garage")
+        })
+    })
+
+    test.describe('check registration fields', ()=>{
+        test("field Name", async({page})=>{
+            const popup = page.locator('app-signup-modal')
+            const name = popup.locator('#signupName')
+            const nameErrorMessage = popup.locator('#signupName + .invalid-feedback')
+            
+            await name.focus()
+            await name.blur()
+
+            await expect(nameErrorMessage).toBeVisible()
+            await expect(nameErrorMessage).toHaveText('Name required')
+            await expect(nameErrorMessage).toHaveCSS('border-color', 'rgb(220, 53, 69)')
+
+            await name.fill(' test')
+            await expect(nameErrorMessage).toHaveText('Name is invalid')
+
+            await name.fill(' ')
+            await expect(nameErrorMessage).toHaveText('Name is invalid'+'Name has to be from 2 to 20 characters long')
```

Comment:
either separate each validation rule by test.step or use dedicated test for each rule

Reply:
it is also related to all next tests